### PR TITLE
[Feature] Merge client

### PR DESCRIPTION
**Pull request by Civolilah**
Here is the screenshot of implemented solution for merging Clients:

Screenshot 2023-01-24 at 14 59 48

### CODE REVIEW

File: src/components/DataTable.tsx [hunk 1]
```diff
@@ -354,9 +346,7 @@ export function DataTable<T extends object>(props: Props<T>) {
                       {props.customActions &&
                         props.customActions?.map(
                           (action: any, index: number) => (
-                            <React.Fragment key={index}>
-                              {action(resource)}
-                            </React.Fragment>
+                            <div key={index}>{action(resource)}</div>
```
Comment (Civolilah) [inline src/components/DataTable.tsx:349]: The reason for changing React.Fragment to a div HTML element is that React.Fragment doesn't have any events like onClick, but we really need it here. We need to close the dropdown after clicking on a particular option. In our case, if we don't do this, the modal and dropdown will open at the same time, which is very bad behavior IMO.

File: src/pages/clients/edit/components/Contacts.tsx [hunk 2]
```diff
@@ -73,7 +73,7 @@ export function Contacts(props: Props) {
         <div key={index} className="pb-4 mb-4 border-b">
           <Element leftSide={t('first_name')}>
             <InputField
-              id="first_name"
+              id={`first_nam_${index}`}
```
Comment (Civolilah) [inline src/pages/clients/edit/components/Contacts.tsx:76]: If we have more than one contact assigned to a particular client, we will get an error in the console that our elements do not have unique IDs in the DOM. It first appeared when I opened the PassowordConfirmation modal.

**Comment by turbo124** [comment]
looks really good :+1:

File: src/components/dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -59,8 +59,17 @@ export function Dropdown(props: Props) {
...
-                ? cloneElement(child, { setVisible, key: index })
+              child &&
+              (child['type'] == DropdownElement || child['type'] == 'div')
+                ? cloneElement(child, {
+                    ...(child['type'] == DropdownElement && {
+                      setVisible,
+                    }),
+                    ...(child['type'] == 'div' && {
+                      onClick: () => setVisible(false),
+                    }),
```
[the remark this answers is not in the document]
Comment (beganovich) [reply src/components/dropdown/Dropdown.tsx:70]: Is it possible to extract this into separate components or something where we can do clean conditionals? These inline things look really bad IMO and we sacrifice a lot of readability for the sake of another component.

File: src/pages/clients/edit/components/Contacts.tsx (hunk 2 above)
Comment (beganovich) [reply src/pages/clients/edit/components/Contacts.tsx:76]: A subtle change, but `first_nam` is a typo.

**Review by beganovich — CHANGES_REQUESTED** [review]
Looks good, few changes needed 👍

File: src/pages/clients/edit/components/Contacts.tsx (hunk 2 above)
Comment (Civolilah) [reply src/pages/clients/edit/components/Contacts.tsx:76]: Resolved!

File: src/components/dropdown/Dropdown.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Civolilah) [reply src/components/dropdown/Dropdown.tsx:70]: Tbh, I've seen many different examples where people conditionally assign properties like this and so I'm doing it like this.

For your recommendation of separating the components I think it is really unnecessary. I just created a function that will return those properties cleaner and that's what we need here. Let me know what you think about this kind of implementation.

**Review by beganovich — APPROVED** [review]
Looks good, thanks 👍

**Pull request merged**